cilium/CFP-25694-move-cilium-cli-to-cilium-repo.md

```diff
@@ -0,0 +1,123 @@
+# CFP-25694: Move cilium/cilium-cli code into cilium/cilium repository
+
+**Code Name: Mega Ketchup**
+
+**SIG: [sig-cilium-cli]**
+
+**Begin Design Discussion:** 2023-05-25
+
+**Cilium Release:** 1.15
+
+**Authors:** Michi Mutsuzaki `<michi@isovalent.com>`
+
+## Summary
+
+Move [cilium/cilium-cli] code into [cilium/cilium] repository.
+
+## Motivation
+
+The main motivation for moving [cilium/cilium-cli] code into [cilium/cilium]
+repository is to minimize the feature development overhead for adding new tests
```

Comment:
I read this as trying to improve development velocity. Won't there be the unintended side-effect of worsening the velocity of cilium-cli? If we require full cilium CI runs for changes that touch just cilium-cli, I think that'd be a source of frustration again.

Reply:
a perfect timing, i was just adding a point in the impact section.

i think this will be the main practical drawback with this CFP. this is a major slowdown in terms of cilium-cli development speed unless we are super careful about defining required tests for various files. i don't really have any good answer for it though, other than to hand wave and say that this will motivate more people to make the cilium CI run faster.

Reply:
I think today there is a hidden cost of cilium-cli development. CLI changes gets eventually tested by cilium/cilium. Discovering breakages in changes after merging leads leads to higher costs.

Reply:
Sure. We run CLI's CI with only a few configurations compared to cilium/cilium's CI (by "CI" I mean GHA workflows), which has quite a few. For example, after releasing the https://github.com/cilium/cilium-cli/releases/tag/v0.14.4 release, some workflows got broken, which required releasing v0.14.5 with https://github.com/cilium/cilium-cli/pull/1660 regardless that we ran some of the CLI CI builds on cilium/cilium (manually editing workflows).

I think @jibi can tell more war stories from his recent effort to add the Egress GW tests.

Reply:
> I think @jibi can tell more war stories from his recent effort to add the Egress GW tests.

what happened in my case is that I added some more tests to the connectivity suite, everything was green in cilium-cli CI but then the moment we switched to the new version in cilium/cilium (we had to bump from v0.13.x to v0.14.y) the e2e workflow broke.

Worth noting it was because of unrelated/previous changes, so we might end up again in a situation where we get changes merged in the CLI and then it's up to the person bumping the version used in e2e to figure out what caused a breakage rather than the original author of the CLI changes

Reply:
> If we require full cilium CI runs for changes that touch just cilium-cli

i think this is much less of a concern now that jenkins is gone. the full CI run takes ~45 minutes maybe, which is comparable to how long cilium-cli CI takes 👀